pkg/rtc/participant.go

```diff
@@ -2692,6 +2718,10 @@ func (p *ParticipantImpl) handleTrackPublished(track types.MediaTrack, isMigrate
 	p.pendingTracksLock.Lock()
 	delete(p.pendingPublishingTracks, track.ID())
 	p.pendingTracksLock.Unlock()
+
+	if !p.hasPendingMigratedTrack() && !p.migratedTracksPublishedPromise.Resolved() {
+		p.migratedTracksPublishedPromise.Resolve(true, nil)
+	}
```

Comment:
this be added to the critical section above? this races and double resolve causes close of closed channel panic